core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -83,7 +85,12 @@ public DefaultTopologyMonitor(InternalDriverContext context) {
     if (closeFuture.isDone()) {
       return CompletableFutures.failedFuture(new IllegalStateException("closed"));
     }
-    return controlConnection.init(true, false);
+    return controlConnection.init(true, reconnectOnInit);
```

Comment:
Brilliant.  I was staring at the code for minutes trying to figure out where you added logic for this, nice to see it really fits in to the existing implementation and doesn't take a lot of code.

Reply:
The existing implementation was for the case when you use a custom topology monitor that doesn't initialize the control connection. If you do a schema refresh later, you need to initialize it, and have it reconnect if it fails. The only difference is that the returned future represented the outcome of the first attempt only, so I changed that to only complete the future when the reconnection is complete. The case I described above now waits on `ControlConnection.firstConnectionAttemptFuture()` instead.